### DISCUSSION TITLE
Error Curiosity

### DISCUSSION THREAD

**Discussion by blaine07** [ZoeyVid/NPMplus#1312] [Q&A] [answered]
Recently installed NPMplus on Unraid Server; seeing a lot of this in logs and wondering if it's something I have done wrong?

`2024/12/16 17:01:35 [alert] 664#664: worker process 7993 exited with fatal code 2 and cannot be respawned
2024/12/16 17:01:35 [alert] 664#664: worker process 8015 exited with fatal code 2 and cannot be respawned
2024/12/16 17:01:35 [alert] 664#664: worker process 8016 exited with fatal code 2 and cannot be respawned
2024/12/16 17:01:35 [alert] 664#664: worker process 8038 exited with fatal code 2 and cannot be respawned
2024/12/16 17:01:35 [alert] 664#664: worker process 8039 exited with fatal code 2 and cannot be respawned
2024/12/16 17:01:35 [alert] 664#664: worker process 8057 exited with fatal code 2 and cannot be respawned
2024/12/16 17:01:35 [alert] 664#664: worker process 8061 exited with fatal code 2 and cannot be respawned
2024/12/16 17:01:35 [alert] 664#664: worker process 8072 exited with fatal code 2 and cannot be respawned
2024/12/16 17:01:35 [alert] 664#664: worker process 8085 exited with fatal code 2 and cannot be respawned
2024/12/16 17:01:35 [alert] 664#664: worker process 8086 exited with fatal code 2 and cannot be respawned
2024/12/16 17:01:35 [alert] 664#664: worker process 8094 exited with fatal code 2 and cannot be respawned
2024/12/16 17:01:35 [alert] 664#664: worker process 8103 exited with fatal code 2 and cannot be respawned
2024/12/16 17:01:35 [alert] 664#664: worker process 8107 exited with fatal code 2 and cannot be respawned
2024/12/16 17:01:35 [alert] 664#664: worker process 8118 exited with fatal code 2 and cannot be respawned
2024/12/16 17:01:35 [alert] 664#664: worker process 8126 exited with fatal code 2 and cannot be respawned
2024/12/16 18:01:35 [alert] 10758#10758: pthread_create() failed (11: Resource temporarily unavailable)
2024/12/16 18:01:35 [alert] 10787#10787: pthread_create() failed (11: Resource temporarily unavailable)
2024/12/16 18:01:35 [alert] 664#664: fork() failed while spawning "worker process" (11: Resource temporarily unavailable)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: sendmsg() failed (9: Bad file descriptor)
2024/12/16 18:01:35 [alert] 664#664: worker process 10758 exited with fatal code 2 and cannot be respawned
2024/12/16 18:01:35 [alert] 664#664: worker process 10787 exited with fatal code 2 and cannot be respawned`

**Comment by blaine07** [comment]
Also saw this at Startup; maybe defining user not necessary on unraid anymore?::

`useradd warning: npm's uid 99 outside of the UID_MIN 1000 and UID_MAX 6000 range.
chown: cannot access '/proc/self/fd/3': No such file or directory
chown: cannot access '/proc/self/fd/4': No such file or directory
chown: cannot access '/proc/self/fd/5': No such file or directory
2024/12/16 18:26:18 [notice] 6112#6112: ModSecurity-nginx v1.0.3 (rules loaded inline/local/remote: 0/7/0)
2024/12/16 18:26:18 [alert] 6115#6115: sched_setaffinity() failed (22: Invalid argument)
2024/12/16 18:26:18 [alert] 6116#6116: sched_setaffinity() failed (22: Invalid argument)
2024/12/16 18:26:18 [alert] 6117#6117: sched_setaffinity() failed (22: Invalid argument)
2024/12/16 18:26:18 [alert] 6120#6120: sched_setaffinity() failed (22: Invalid argument)
removed '/data/tls/certbot/archive/npm-2/cert1.pem'
removed '/data/tls/certbot/archive/npm-2/privkey1.pem'
removed '/data/tls/certbot/archive/npm-2/chain1.pem'
removed '/data/tls/certbot/archive/npm-2/fullchain1.pem'
'/usr/local/nginx/conf/conf.d/include/coreruleset/plugins/README.md' -> '/data/etc/modsecurity/crs-plugins/README.md'
'/usr/local/nginx/conf/conf.d/include/coreruleset/plugins/empty-after.conf' -> '/data/etc/modsecurity/crs-plugins/empty-after.conf'
'/usr/local/nginx/conf/conf.d/include/coreruleset/plugins/empty-before.conf' -> '/data/etc/modsecurity/crs-plugins/empty-before.conf'
'/usr/local/nginx/conf/conf.d/include/coreruleset/plugins/empty-config.conf' -> '/data/etc/modsecurity/crs-plugins/empty-config.conf'
no DEFAULT_CERT_ID set, using dummycerts.
Working on file: /data/nginx/default.conf
Working on file: /data/nginx/proxy_host/1.conf
Success.`

**Comment by Zoey2936** [comment]
Please try with a uid within the supported range

**Reply by blaine07** [reply]
On Unraid do you have any idea what that could be or how I’d find it?

**Reply by Zoey2936** [reply]
you set the env to a supported value?

**Comment by blaine07** [comment]
![Screenshot 2024-12-17 150049](https://github.com/user-attachments/assets/3f30df37-4a0a-4b82-9186-1f43b05efbb1)
![Screenshot 2024-12-17 150042](https://github.com/user-attachments/assets/43ea4494-1ed7-447d-ab66-33bdea24bdb9)

They're set as this; just unsure if necessary or correct for Unraid?

**Reply by Zoey2936** [reply]
`useradd warning: npm's uid 99 outside of the UID_MIN 1000 and UID_MAX 6000 range` do you read your own error logs?

**Reply by blaine07** [reply]
I guess I didin't understand that's what it meant.

IS this '[alert] 741#741: sched_setaffinity() failed (22: Invalid argument)' related to UID error or can you tell me what it means exactly? Unsuccessfully tried Google, and read the error this time; no idea what setaffinity is.

**Reply by Zoey2936** [reply, marked as answer]
please set the uid and guid to something higher then (or equal to) 1000
